src/rpc.rs

```diff
@@ -376,3 +379,51 @@ pub fn get_id_specific_size(num: Block) -> HashMap<u32, u32> {
 	}
 	index
 }
+
+pub async fn check_connection(
+	full_node_ws: Vec<String>,
+) -> Result<Option<WebSocketStream<MaybeTlsStream<TcpStream>>>> {
+	let mut ws_ = None;
+	for x in full_node_ws.iter() {
+		let _url = url::Url::parse(&x)?;
```

Comment:
If we parse URLs before calling this method, error path would be clear and this method would be simpler (instead of result, it would return `Option<WebSocketStream<MaybeTlsStream<TcpStream>>>`